File: php/hamming/hamming.php
<?php

// It is found by comparing two DNA strands and counting how many of the nucleotides are different from their equivalent in the 
// other string

function distance($a, $b)
{
    if (strlen($a) !== strlen($b)) {
        throw new InvalidArgumentException('DNA strands must be of equal length.');
    }
    $a_array = str_split($a);
    $b_array = str_split($b);
    return count(array_diff_assoc($a_array, $b_array));
}

File: ruby/hamming/hamming.rb
# It is found by comparing two DNA strands and counting how many of the
# nucleotides are different from their equivalent in the other string
class Hamming
  def self.compute(a_str, b_str)
    message = 'DNA strands must be of equal length.'
    raise ArgumentError, message unless a_str.length == b_str.length

    a_str.chars.zip(b_str.chars).reject { |a_val, b_val| a_val == b_val }.count
  end
end
